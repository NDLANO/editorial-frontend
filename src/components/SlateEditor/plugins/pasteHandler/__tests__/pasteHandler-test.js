import { Block } from 'slate';
import { convertToSupportedBlockTypes } from '..';

const lineBlock = {
  data: {},
  nodes: [
    {
      leaves: [
        {
          marks: [],
          object: 'leaf',
          text: 'Text from PlaintextEditor',
        },
      ],
      object: 'text',
    },
  ],
  object: 'block',
  type: 'line',
};

test('Paste transforms plaintext line block type to paragraph type', () => {
  const pastedBlock = Block.fromJSON(lineBlock);
  const convertedBlock = convertToSupportedBlockTypes(pastedBlock);
  expect(convertedBlock.toJSON()).toMatchSnapshot();
});
