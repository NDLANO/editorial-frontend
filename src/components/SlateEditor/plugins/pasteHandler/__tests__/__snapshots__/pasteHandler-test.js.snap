// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paste transforms plaintext line block type to paragraph type 1`] = `
Object {
  "data": Object {},
  "nodes": Array [
    Object {
      "marks": Array [],
      "object": "text",
      "text": "Text from PlaintextEditor",
    },
  ],
  "object": "block",
  "type": "paragraph",
}
`;
